lib/driver.rb

```diff
@@ -43,5 +44,67 @@ def add_trip(trip)
 
       @trips << trip
     end
+
+    def get_total_revenue
+      total_rev = 0.0
+      @trips.each do |trip|
+        if trip.cost != nil
+          trip_cost = (trip.cost - 1.65) * 0.8
+          total_rev += trip_cost.to_f
+        end
+      end
+      return total_rev.round(2)
+    end
+
+    def total_hours_driving
+      total_seconds = 0.0
+      @trips.each do |trip|
+        if trip.end_time != nil
+          trip_time = trip.end_time - trip.start_time
+          total_seconds += trip_time.round(2)
+        end
+      end
+      total_hours = total_seconds / 3600
+      return total_hours.round(2)
+    end
+
+    def avg_revenue_per_hour
+      if last_trip_not_nil?
+        avg_rev_per_hour = get_total_revenue / total_hours_driving
+      end
+      return avg_rev_per_hour.round(2)
```

Comment:
When you run `rake` to run the tests, you end up getting an error saying that `avg_rev_per_hour` is `nil`. That's because there is no value assigned to `avg_rev_per_hour`... there is a value assigned to it IF `last_trip_not_nil?` evaluates to true. Otherwise it's nil!